proxy/grpcproxy/watch_broadcasts.go

```diff
@@ -96,7 +96,8 @@ func (wbs *watchBroadcasts) add(w *watcher) {
 	wbs.bcasts[wb] = struct{}{}
 }
 
-func (wbs *watchBroadcasts) delete(w *watcher) {
+// delete removes a watcher and returns the number of remaining watchers.
+func (wbs *watchBroadcasts) delete(w *watcher) int {
```

Comment:
yeah, it's not great but it seemed better than having empty() reacquire the lock just to calm down the race detector